pylint/checkers/variables.py

```diff
@@ -1221,7 +1251,14 @@ def visit_name(self, node: nodes.Name) -> None:
                                 "undefined-variable", node=node, args=node.name
                             )
 
-            current_consumer.mark_as_consumed(node.name, found_nodes)
+            # Some nodes were filtered out because they execute only on success
```

Comment:
I don't particularly like how this works, as any new contributor needs to read about this to understand this. It is not as self-explanatory as `to_consume` and `consumed`. However, I don't know of a better way to do this. Perhaps we could improve on the name? I'm not sure.. I'll think about this some more.

Reply:
How about `consumed_conditionally`?

Reply:
Either works for me. @Pierre-Sassoulas do you have an opinion?

Reply:
'maybe_consumed' ? I don't have en opinion either :)